src/RemoteInboxNotifications/RuleEvaluator.php

```diff
@@ -42,15 +45,33 @@ public function evaluate( $rules, $stored_state = null ) {
 			return false;
 		}
 
+		$evaluation_logger = null;
+
+		if ( count( $logger_args ) ) {
+			if ( ! array_key_exists( 'slug', $logger_args ) ) {
+				throw new \InvalidArgumentException( 'Missing required field: slug in $logger_args.' );
```

Comment:
Throwing `InvalidArgumentException` here to let the dev know that there is something wrong immediately since we only use the logger during development. I'm open to just error to `debug.log` though.

Reply:
This would make `slug` required for all implementations using the rule evaluator.  If we turn on the debug flag, all instances that have not implemented `slug` will begin to fail in development and may be hard to work with.

I'm okay with making this mandatory if we think it's really important, but we'll need to audit all uses of the rule evaluator.

It would be great instead to dynamically create a slug based on the rules provided, but I'm guessing this could result in some pretty long slugs.

Reply:
I think the existing implementations should be okay since it only throws the exception when `$logger_args` has at least one value in it. I think `slug` is required since we need a unique identifier when viewing the log.